server/src/main/scala/com/cloudera/livy/server/interactive/InteractiveSession.scala

```diff
@@ -70,36 +71,48 @@ class InteractiveSession(
     val conf = prepareConf(request.conf, request.jars, request.files, request.archives,
       request.pyFiles)
 
-    info(s"Creating LivyClient for sessionId: $id")
-    val builder = new LivyClientBuilder()
-      .setConf("spark.app.name", s"livy-session-$id")
-      .setAll(conf.asJava)
-      .setConf("livy.client.sessionId", id.toString)
-      .setConf(RSCConf.Entry.DRIVER_CLASS.key(), "com.cloudera.livy.repl.ReplDriver")
-      .setURI(new URI("rsc:/"))
+    val builderProperties = mutable.Map[String, String]()
+    builderProperties ++= conf
 
     def mergeConfList(list: Seq[String], key: String): Unit = {
       if (list.nonEmpty) {
-        val newList = (list ++ conf.get(key)).mkString(",")
-        builder.setConf(key, newList)
+        builderProperties.get(key) match {
+          case None =>
+            builderProperties.put(key, list.mkString(","))
+          case Some(oldList) =>
+            val newList = (oldList :: list.toList).mkString(",")
+            builderProperties.put(key, newList)
+        }
+      }
+    }
+
+    def mergeHiveSiteAndHiveDeps(): Unit = {
+      hiveSiteFile(livyConf) match {
```

Comment:
This checks the Livy configuration, but not the user's session config; in yarn-cluster mode, `--files` can be used to distribute a custom hive-site.xml. Since it's a little tricky to figure out the master configuration here, and it really only works in yarn-cluster mode, I think it's ok to ignore that for now.